### PR TITLE
Modernise and repeat CDI Annotations tests

### PR DESCRIPTION
**Pull request by tevans78**
At the same time as allowing these tests to repeat against JEE 9, I have taken the opportunity to update them a little.

### CODE REVIEW

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#run-libby-bot

File: dev/com.ibm.ws.cdi.api_fat/test-applications/alterableContextApp.war/src/com/ibm/ws/cdi12/alterablecontext/test/AlterableContextTestServlet.java [hunk 1]
```diff
@@ -33,12 +33,14 @@
     private static String SECOND = "Now the command returns: null";
 
     @Test
-    public void testBeanWasFound(HttpServletRequest request, HttpServletResponse response) throws Exception {
+    @Mode(TestMode.FULL)
+    public void testBeanWasFound() throws Exception {
         List<String> strings = DirtySingleton.getStrings();
         assertContains(strings, FIRST);
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.api_fat/test-applications/alterableContextApp.war/src/com/ibm/ws/cdi12/alterablecontext/test/AlterableContextTestServlet.java:39]: `assertThat(strings, hasItem(containsString(FIRST)))`

The current `assertContains` method does not appear to output the actual value.

File: dev/com.ibm.ws.cdi.api_fat/test-applications/injectInjectionPointAsParam.war/src/com/ibm/ws/fat/cdi/injectInjectionPointParam/InjectInjectionPointAsParamServlet.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Inject
+    TestBean bean;
+
+    @Test
+    @Mode(TestMode.FULL)
+    public void testInjectInjectionPointAsParam() throws IOException {
+        if (!bean.test()) {
+            fail("Bean Manager and/or Injection Point not correctly injected into TestBean constructor");
+        }
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.api_fat/test-applications/injectInjectionPointAsParam.war/src/com/ibm/ws/fat/cdi/injectInjectionPointParam/InjectInjectionPointAsParamServlet.java:39]: Probably better to do asserts inside `TestBean.test()` so you have a better idea of what actually failed.

**Pull request merged**